libsolidity/ast/Types.cpp

```diff
@@ -561,6 +804,55 @@ shared_ptr<IntegerType const> IntegerConstantType::integerType() const
 		);
 }
 
+shared_ptr<FixedPointType const> RationalNumberType::fixedPointType() const
+{
+	bool negative = (m_value < 0);
+	unsigned fractionalBits = 0;
+	unsigned integerBits = 0;
+	rational value = m_value;
+	bigint transitionValue = bigint(1) << 256;
+	rational maxValue = rational(transitionValue);
+
+	if (!negative)
+	{
+		maxValue -= 1;
+		integerBits = bytesRequired(wholeNumbers()) * 8;
+	}
+	else
+	{
+		value = abs(value);
+		if (wholeNumbers() > 0)
+			transitionValue = ((0 - wholeNumbers()) - 1) << 1;
+		else
+			transitionValue = 0;
+		integerBits = bytesRequired(transitionValue) * 8;
+	}
+
+	while (value * 0x100 <= maxValue && value.denominator() != 1 && fractionalBits < 256 - integerBits)
```

Comment:
Also I put the integer bits in the while loop to make the loop more efficient. We go until we can't fit anymore and enables more numbers to get into the system that users would like to use. This is why we need it before we get the value.